### PR TITLE
[Table] Frozen Rows/Cols - Part 8: Fix tests

### PR DESCRIPTION
**Pull request by cmslewis**
#### Checklist

- [x] Include tests

#### Changes proposed in this pull request:

- Fix tests that I broke and `skip`'d in past PRs.
    - In particular, be sure to `delayToNextFrame` when testing anything to do with scrolling, which is now throttled via `requestAnimationFrame`.
    - Also, test the `MAIN` quadrant's scroll container by default.

#### Reviewers should focus on:

- ~HOW DO I FIX THE BROKEN `TruncatedFormat` test? Can't figure it out.~ Fixed with `position: relative;`. :(
    - https://github.com/palantir/blueprint/blob/cl/table-frozen-rows-cols-8-fix-tests/packages/table/test/formatsTests.tsx#L55

### CODE REVIEW

**Comment by cmslewis** [comment]
Ughhh and now I'm seeing the same test flaking out on `master`:
image

File: packages/table/test/formatsTests.tsx [hunk 1]
```diff
@@ -69,7 +71,7 @@ describe("Formats", () => {
                 majority have never stirred?
             `;
 
-            const style = { height: "200px" };
+            const style = { height: "200px", position: "relative" };
```
Comment (themadcreator) [inline packages/table/test/formatsTests.tsx]: i dont entirely understand why this would have broken or why this change fixes it but it's possible some deeply nested SCSS rule became invalid with the new DOM hierarchy.

if so, we should add this css rule to the TABLE_TRUNCATED_TEXT class instead of just the style on the test

Comment (cmslewis) [inline packages/table/test/formatsTests.tsx]: > if so, we should add this css rule to the TABLE_TRUNCATED_TEXT class instead of just the style on the test

that's what i was wondering. lemme test it out locally.

Comment (cmslewis) [inline packages/table/test/formatsTests.tsx]: NOPE. That would be bad.



@themadcreator would you prefer I find the exact reason, or just leave the fix here, merge this PR, and revisit? I'm leaning toward the former, but if I add `position: relative`, quite a few downstream CSS fixes are required too. Might be out of scope for now.
